src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractHpiMojo.java

```diff
@@ -827,31 +827,36 @@ private static void copyFilteredFile(File from, File to, String encoding, Filter
     /**
      * If the project is on Git, figure out Git SHA1.
      *
-     * @return null if no git repository is found
-     */
-    public String getGitHeadSha1() {
-        // we want to allow the plugin that's not sitting at the root (such as findbugs plugin),
-        // but we don't want to go up too far and pick up unrelated repository.
-        File git = new File(project.getBasedir(), ".git");
```

Comment:
this was questionable to begin with.
it implies that
1. you can only ever have a 1 level structure
2. it would be valid and useful to checkout a non git project into a clone of a git projects directory structure (but only at 2 levels or lower!)